### PR TITLE
Actualize 'Authoring Theia Extensions' doc

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

Adjusted `Authoring_Extensions.md` to the actual code generated by `generator-theia-extension`. Also corrected some inaccuracies.

### CODE REVIEW

File: doc/Authoring_Extensions.md [hunk 1]
```diff
@@ -79,9 +77,9 @@ We also got a `lerna.json` file to configure `lerna`:
 
 ## Implementing the Extension
 
-Next let's look at the generated code for our extension in the `hello-world-extension` folder. Let’s start with the `package.json`. It specifies the package’s metadata, its dependencies to the (bleeding edge) theia core package, a few scripts and dev dependencies, and the theia-extensions.
+Next let's look at the generated code for our extension in the `hello-world-extension` folder. Let’s start with the `package.json`. It specifies the package’s metadata, its dependencies to the (bleeding edge) Theia core package, a few scripts and dev dependencies, and the theia-extensions.
 
-The keyword `theia-extension` is important: It allows a Theia app to identify and install Theia extensions from `npm`.
+The keyword `theiaExtensions` is important: It allows a Theia app to identify and install Theia extensions from `npm`.
```
Comment (akosyakov) [inline doc/Authoring_Extensions.md]: it was correct, the keyword is `theia-extension`, e.g.: https://github.com/theia-ide/theia/blob/55e431bd0ebff81a64a6ef1e53edc74dfb815bf6/packages/process/package.json#L18-L20

**Comment by kittaakos** [comment]
@azatsarynnyy, thanks for the update!

Since you have corrected the documentation, could you please fix the typo in the _defines one ore more dependency_ too? `ore` -> `or`

File: doc/Authoring_Extensions.md (hunk 1 above)
Comment (azatsarynnyy) [reply doc/Authoring_Extensions.md]: @akosyakov reverted

**Comment by azatsarynnyy** [comment]
@kittaakos fixed

**Comment by akosyakov** [comment]
@JanKoehnlein can you review since you authored it

**Comment by azatsarynnyy** [comment]
I squashed the commits and PR approve was dismissed.
Could you guys approve it again, please? Sorry for disturbing.

**Pull request merged**